raster_analysis/tiling.py

```diff
@@ -1,144 +1,143 @@
+from decimal import Decimal
 from shapely.geometry import mapping, box
-
-import boto3
-import json
 import logging
 import os
-
-import threading
-import queue
-
+from datetime import date
+from time import sleep
 from copy import deepcopy
-
-import numpy as np
-
-from raster_analysis.exceptions import RasterAnalysisException
+from raster_analysis.boto import dynamodb_resource
+import pandas as pd
 from aws_xray_sdk.core import xray_recorder
+from raster_analysis.boto import lambda_client
 
 logger = logging.getLogger(__name__)
 logger.setLevel(logging.INFO)
 
-RASTER_ANALYSIS_LAMBDA_NAME = os.environ["RASTER_ANALYSIS_LAMBDA_NAME"]
-
 
 @xray_recorder.capture("Merge Tiled Geometry Results")
 def merge_tile_results(tile_results, groupby_columns):
-    concatted_tile_results = concat_tile_results(tile_results)
-
-    groupby_results = np.array([concatted_tile_results[col] for col in groupby_columns])
-    column_maxes = [col.max() + 1 for col in groupby_results]
-    linear_indices = np.ravel_multi_index(groupby_results, column_maxes).astype(
-        np.uint32
-    )
-
-    unique_values, inv = np.unique(linear_indices, return_inverse=True)
-    unique_value_combinations = np.unravel_index(unique_values, column_maxes)
-
-    merged_tile_results = dict(zip(groupby_columns, unique_value_combinations))
-
-    for col_name, col_data in concatted_tile_results.items():
-        if col_name not in groupby_columns:
-            arr = np.array(col_data)
-            merged_tile_results[col_name] = np.bincount(inv, weights=col_data).astype(
-                arr.dtype
+    if not groupby_columns:
+        dataframes = [pd.DataFrame(result, index=[0]) for result in tile_results]
+        merged_df: pd.DataFrame = pd.concat(dataframes)
+        return merged_df.sum().to_dict()
+
+    dataframes = [pd.DataFrame(result) for result in tile_results]
+    merged_df: pd.DataFrame = pd.concat(dataframes)
+
+    grouped_df: pd.DataFrame = merged_df.groupby(groupby_columns).sum()
```

Comment:
why not use the custom numpy group_by function you wrote?

Reply:
I decided recently to just use pandas here. I was using NumPy in the main raster analysis lambda for grouping because the pandas one was consuming a ton of memory. But that doesn't matter in this lambda, and it made the code cleaner and gave access to more functions like easy concatenation, sorting, and converting to a JSON-like structure.